core/src/main/scala/io/chrisdavenport/natchezlocal/TraceLocal.scala

```diff
@@ -27,6 +29,39 @@ object TraceLocal {
 
       def traceUri =
         local.get.flatMap(_.traceUri)
-    }
 
-}
\ No newline at end of file
+      override def log(fields: (String, TraceValue)*): F[Unit] =
+        local.get.flatMap(_.log(fields: _*))
+
+      override def log(event: String): F[Unit] =
+        local.get.flatMap(_.log(event))
+
+      override def attachError(err: Throwable): F[Unit] =
+        local.get.flatMap(_.attachError(err))
+
+      override def spanR(name: String, kernel: Option[Kernel]): Resource[F, F ~> F] =
+        Resource {
+          local.get.flatMap { parent =>
+            kernel.fold(parent.span(name))(parent.span(name, _))
+              .allocated
+              .map {
+                _.leftMap { child =>
+                  new (F ~> F) {
+                    override def apply[A](fa: F[A]): F[A] =
+                      (local.set(child) >> fa.onError {
+                        case e => child.attachError(e)
+                      }).guarantee(local.set(parent))
```

Comment:
I'm not sure if this is right, see https://github.com/tpolecat/natchez/pull/679.

Reply:
Thanks, I updated it based on what the `IOLocal`-based `Trace` instance does. Can you take another look?

Reply:
Thanks that looks good! :)